cs3api4lab/tests/share_test_base.py

```diff
@@ -188,3 +190,37 @@ def remove_test_file(self, user, file_path):
 
     def get_random_suffix(self):
         return '-' + ''.join(random.choices(string.ascii_lowercase + string.digits, k=5))
+
+    def clean_up_file(self, user, file_name):
+        try:
+            self.remove_test_file(user, file_name)
+        except Exception as ex:
+            print(str(ex))
+
+    def clean_up_share(self, user, share_id):
+        try:
+            self.remove_test_share(user, share_id)
+        except Exception as ex:
+            print(str(ex))
+
+    def remove_share_and_file_by_path(self, user, file_path):
+        if user == 'einstein':
+            share_api = self.share_api
+            storage = self.storage_logic
+        if user == 'marie':
+            share_api = self.marie_share_api
+            storage = self.marie_storage_logic
+        if user == 'richard':
+            share_api = self.richard_share_api
+            storage = self.richard_storage_logic
+
+        stat = storage.stat(file_path, None)
+        if stat is None:
+            self.create_test_file(user, file_path)
+
+        shares = share_api.list_grantees_for_file(file_path) #todo this won't work on CERNBOX
```

Comment:
because on default reva config share id includes filepath, on Cernbox it's just an id

Reply:
more exactly share.resource_id.opaque_id

Reply:
It's true that in CERNBox we use the IDs in the opaque_id, but in general the cs3apis does not specify if the content is a path. From the [doc](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.ResourceId):
`The internal id used by service implementor to uniquely identity the resource in the internal implementation of the service.`

This could be also a path (as it's unique), but it's not guaranteed. To make it compliant with the specs, you should use directly the resource_id.